apollo-spaceport/build.rs

```diff
@@ -1,12 +1,54 @@
-fn main() {
+use std::{
+    error::Error,
+    fs::File,
+    io::{copy, Read},
+};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    // Retrieve a live version of the reports.proto file
+    let proto_url = "https://usage-reporting.api.apollographql.com/proto/reports.proto";
+    let response = reqwest::blocking::get(proto_url)?;
+    let mut content = response.text()?;
+
+    // Process the retrieved content to:
+    //  - Insert a package Report; line after the import lines (currently only one) and before the first message definition
+    //  - Remove the Apollo TS extensions [(js_use_toArray)=true] and [(js_preEncoded)=true] from the file
+    //  Note: Only two in use at the moment. This may fail in future if new extensions are
+    //  added to the source, so be aware future self. It will manifest as a protobuf compile
+    //  error.
+    let message = "\nmessage";
+    let msg_index = content.find(message).ok_or("cannot find message string")?;
+    content.insert_str(msg_index, "\npackage Report;\n");
+
+    content = content.replace("[(js_use_toArray)=true]", "");
+    content = content.replace("[(js_preEncoded)=true]", "");
+
+    // Try to avoid writing out the same content since it will trigger unnecessary re-builds, which wastes time
```

Comment:
is this to prevent triggering the tonic build script?

Reply:
yes